Suggest similar investigations based on the currently displayed investigation on investigation landing page

kennethnym opened the pull request:
## Description
This PR adds a section on investigation landing pages that suggests similar investigations based on the currently displayed investigations. Investigation suggestions and extra info for the investigation are grouped into 2 accordions for better visual clarity. This video demonstrates the new UI:


https://user-images.githubusercontent.com/19359984/233374428-8d1c78f0-f6e1-46aa-88eb-0516023f35c4.mp4

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
Closes #1521

kennethnym commented:
Absolutely, do you have a particular number in mind? Obviously we don't want to show too much items, otherwise it'll crowd the page too much. Is 5 a good number to you?

agbeltran commented:
maybe let's show 10 - or @saifulkhan can comment on how many would be better to show at this stage (notice that this is not the final UI, it is just for demo purposes)

kennethnym commented:
Addressing feedback provided by @saifulkhan, I have added topic chips & pagination to the suggestion list. Demo below:

https://user-images.githubusercontent.com/19359984/234890482-e98afd44-155c-4a2c-b757-6519ef67400a.mp4

I've updated the original comment as well with the new demo.

kennethnym commented:
Placing the pagination control at the bottom causes it to be unstable, i.e. the control doesn't stay in the same position when switching between pages, which means the users would have to re-aim the mouse cursor between page changes.

louise-davies reviewed (commented):
Code looks good to me, maybe just add a TODO comment on the hardcoded URL to remind us for later that it needs to be changed. Only commenting for now instead of approving because we're awaiting a demo of this feature which would unblock "doing it properly" e.g. not hardcoding the URL

louise-davies commented on packages/datagateway-dataview/src/views/landing/isis/isisInvestigationLanding.component.tsx:
```diff
@@ -415,27 +420,38 @@ const LandingPage = (props: LandingPageProps): React.ReactElement => {
+                            {field.content(data[0] as Investigation)}
+                          </ShortInfoValue>
+                        </ArrowTooltip>
+                        z
```
Remove this random `z` character

kaperoo replied:
I thought the z character was intentional 😅

louise-davies reviewed (commented):
I think I prefer the look of the links - it makes it much clearer that they're clickable and that they're links to other web pages. Theoretically, you could keep the whole button clickable and just make the text "look" like a link, but tbh I think it's just easier to have them as links.

But otherwise, I really like the chips to show the similarity score and the functionality works well :)

kaperoo commented:
> I think I prefer the look of the links - it makes it much clearer that they're clickable and that they're links to other web pages. Theoretically, you could keep the whole button clickable and just make the text "look" like a link, but tbh I think it's just easier to have them as links.
> 
> But otherwise, I really like the chips to show the similarity score and the functionality works well :)

Changed the buttons back to links and removed the random 'z' in extra info.

louise-davies commented:
@VKTB this is ready from our end again
